### ISSUE TITLE
When adding new datapoints they do not show on the map

### ISSUE THREAD

**Issue by valllllll2000** [akvo/akvo-flow-mobile#728]
Steps to reproduce:
Create survey with location question and select "use as datapoint location".
Start adding datapoints
You will see that they do not /not all appear on the map
Killing the app and restarting displays the datapoints correctly
This should probably be fixed once the datapoint refactor is merged #583

**Comment by janagombitova** [comment]
Data points on version 2.4.0 add added to the map (if geolocation is given) after being submitted as they should. Another issue tackled 👍  Nice
